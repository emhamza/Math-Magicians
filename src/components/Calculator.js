import React, { useState } from 'react';
import ButtonBox from './ButtonBox';
import Button from './Button';
import calculate from '../logics/calculate';

const btnValues = [
  [{ id: 'AC' }, { id: '+/-' }, { id: '%' }, { id: 'รท' }],
  [{ id: '7' }, { id: '8' }, { id: '9' }, { id: 'x' }],
  [{ id: '4' }, { id: '5' }, { id: '6' }, { id: '-' }],
  [{ id: '1' }, { id: '2' }, { id: '3' }, { id: '+' }],
  [{ id: '0' }, { id: '.' }, { id: '=' }],
];

const Calculator = () => {
  const [obj, setObj] = useState({
    total: '0',
    next: null,
    operation: null,
  });

  const handleEvent = (btn) => setObj(calculate(obj, btn));

  return (
    <>
      <h3>Lets do some Math!</h3>
      <div className="calculator">
        <div className="screen">
          <p>
            {obj?.total}
            {' '}
            {obj?.operation}
            {' '}
            {obj?.next}
          </p>
        </div>
        <ButtonBox>
          {btnValues.flat().map((btn) => (
            <Button
              onClick={() => handleEvent(btn.id)}
              value={btn.id}
              key={btn.id}
            />
          ))}
        </ButtonBox>
      </div>
    </>
  );
};

export default Calculator;
