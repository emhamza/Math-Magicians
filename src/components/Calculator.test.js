import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react';
import Calculator from './Calculator';

describe('Calculator', () => {
  it('should render the component', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should update the screen on button click', () => {
    const { getByTestId, getByText } = render(<Calculator />);
    fireEvent.click(getByText('2'));
    fireEvent.click(getByText('+'));
    fireEvent.click(getByText('3'));
    fireEvent.click(getByText('='));
    expect(getByTestId('screen')).toHaveTextContent('5');
  });
});
