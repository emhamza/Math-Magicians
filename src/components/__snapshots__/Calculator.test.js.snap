// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders correctly 1`] = `
<DocumentFragment>
  <h3>
    Lets do some Math!
  </h3>
  <div
    class="calculator"
  >
    <div
      class="screen"
    >
      <p>
        0  
      </p>
    </div>
    <div
      class="buttonBox"
    >
      <button
        class="undefined button"
        type="submit"
      >
        AC
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        +/-
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        %
      </button>
      <button
        class="operator button"
        type="submit"
      >
        รท
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        7
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        8
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        9
      </button>
      <button
        class="operator button"
        type="submit"
      >
        x
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        4
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        5
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        6
      </button>
      <button
        class="operator button"
        type="submit"
      >
        -
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        1
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        2
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        3
      </button>
      <button
        class="operator button"
        type="submit"
      >
        +
      </button>
      <button
        class="bigbtn button"
        type="submit"
      >
        0
      </button>
      <button
        class="undefined button"
        type="submit"
      >
        .
      </button>
      <button
        class="operator button"
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Calculator should render the component 1`] = `
<h3>
  Lets do some Math!
</h3>
`;
